### DISCUSSION TITLE
Birdnet-Pi on a USB key or SD card

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#750] [Q&A]
Hello,

What is the best solution regarding speed, stability, longevity? Installing Birdnet-Pi on a USB key or SD card?

Thank you.

**Comment by norfolk-fr** [comment]
Hi,
My setup is a mix of the sd and usb : 
Birdnet core scripts run at the sd card level.
All Processed data are in /dev/shm to preserve the sd card.
All Extracted data are stored in a multi device drive (2 usb stick in a raid1 array)
All logs files (/var/log) are managed with log2ram with 128M size virtual hdd to preserve the sd card.

CU !
Alban

**Reply by blogdemoi** [reply]
Thank you. Do you think installing everything (as I did) on a usb key can reduce speed or render the system less effective?

**Reply by norfolk-fr** [reply]
Hi,
I'd say that depends of your hardware. I think it can work with a raspberry pi 4 because of the usb 3 support.
I would not recommend a full install on a usb key with a raspberry pi 3.
How does it work on you side with this installation ?

**Reply by blogdemoi** [reply]
> How does it work on you side with this installation ?

The usb key installation is working fine. My problem is that the Pi 3B seems not to be powerful enough for the Birdet-Pi system (I even overclocked my Pi a little), and detections lags behind a few hours at the end of the day...
